packages/react-next/src/components/Image/Image.base.tsx

```diff
@@ -0,0 +1,173 @@
+  }, [loadState]);
+
+  const onImageLoaded = React.useCallback(
+    (ev: React.SyntheticEvent<HTMLImageElement>) => {
+      props.onLoad?.(ev);
+      if (props.src) {
+        setLoadState(ImageLoadState.loaded);
+      }
+    },
+    [props.src, props.onLoad],
+  );
+
+  const onImageError = React.useCallback(
+    (ev: React.SyntheticEvent<HTMLImageElement>) => {
+      props.onError?.(ev);
+      setLoadState(ImageLoadState.error);
+    },
+    [props.src, props.onError],
+  );
+
+  return [loadState, onImageLoaded, onImageError] as const;
+}
+
+export const ImageBase = React.forwardRef((props: IImageProps, forwardedRef: React.Ref<HTMLImageElement>) => {
+  const frameElement = React.useRef<HTMLDivElement>() as React.RefObject<HTMLDivElement>;
+  const imageElement = React.useRef<HTMLImageElement>() as React.RefObject<HTMLImageElement>;
+  const [loadState, onImageLoaded, onImageError] = useLoadState(props, imageElement, frameElement);
+
+  const imageProps = getNativeProps<React.ImgHTMLAttributes<HTMLImageElement>>(props, imgProperties, [
+    'width',
+    'height',
+  ]);
+  const {
+    src,
+    alt,
+    width,
+    height,
+    shouldFadeIn = true,
+    shouldStartVisible,
+    className,
+    imageFit,
+    role,
+    maximizeFrame,
+    styles,
+    theme,
+  } = props;
+  const coverStyle = computeCoverStyle(props, loadState, imageElement, frameElement);
+  const classNames = getClassNames(styles!, {
+    theme: theme!,
+    className,
+    width,
+    height,
+    maximizeFrame,
+    shouldFadeIn,
+    shouldStartVisible,
+    isLoaded:
+      loadState === ImageLoadState.loaded || (loadState === ImageLoadState.notLoaded && props.shouldStartVisible),
+    isLandscape: coverStyle === ImageCoverStyle.landscape,
+    isCenter: imageFit === ImageFit.center,
+    isCenterContain: imageFit === ImageFit.centerContain,
+    isCenterCover: imageFit === ImageFit.centerCover,
+    isContain: imageFit === ImageFit.contain,
+    isCover: imageFit === ImageFit.cover,
+    isNone: imageFit === ImageFit.none,
+    isError: loadState === ImageLoadState.error,
+    isNotImageFit: imageFit === undefined,
+  });
+
+  // If image dimensions aren't specified, the natural size of the image is used.
+  return (
+    <div className={classNames.root} style={{ width: width, height: height }} ref={frameElement}>
+      <img
+        {...imageProps}
+        onLoad={onImageLoaded}
+        onError={onImageError}
+        key={KEY_PREFIX + props.src || ''}
+        className={classNames.image}
+        ref={useMergedRefs(imageElement, forwardedRef)}
+        src={src}
+        alt={alt}
+        role={role}
+      />
+    </div>
+  );
+});
+ImageBase.displayName = 'ImageBase';
+
+function computeCoverStyle(
+  props: IImageProps,
+  loadState: ImageLoadState,
+  imageElement: React.RefObject<HTMLImageElement>,
+  frameElement: React.RefObject<HTMLDivElement>,
+): ImageCoverStyle {
+  const { imageFit, width, height } = props;
```

Comment:
That would require converting the `else if` to a separate `if` statement. It might be functionally the same with the return, but I feel it's cleaner this way.

Reply:
i meant by move 
`const { imageFit, width, height } = props;` after the `props.coverStyle !== undefined` check :)

```
  if (props.coverStyle !== undefined) {
    return props.coverStyle;
  } else if ... {
   const { imageFit, width, height } = props;
...
}

Reply:
Those values are used in the `else-if` conditional.